src/cpu/cgroup_v2.rs

```diff
@@ -7,9 +7,29 @@ use std::path::Path;
 const CPU_SYS_V2_NUMBER_OF_FIELDS: usize = 3;
 
 #[cfg(target_os = "linux")]
-pub fn read_and_parse_v2_sys_stat(path: &Path) -> Result<CgroupCpuMeasurement> {
-    let time = precise_time_ns();
+pub fn read_and_parse_v2_sys_stat(
+    path: &Path,
+    cpu_max_path: &Path,
+) -> Result<CgroupCpuMeasurement> {
+    // If the cpu.max file exists, we can use it to calculate the number of CPUs
+    // in the cgroup. It's also required that the first value is not set to "max",
+    // otherwise we can't calculate the number of CPUs.
+    let mut cpu_count = 0;
+    if cpu_max_path.exists() {
+        let reader = file_to_buf_reader(&cpu_max_path)?;
+        for line in reader.lines() {
+            let line = line.map_err(|e| ProbeError::IO(e, path_to_string(path)))?;
+            let segments: Vec<&str> = line.split_whitespace().collect();
+            let max = segments[0];
+            let period = parse_u64(&segments[1])?;
+
+            if max != "max" {
+                cpu_count = parse_u64(&max)? / period;
```

Comment:
I think `cpu_count` will be an `u64`. This won't do the right thing when allocating fractional CPU capability, which is apparently something you can do:

```
> docker run --rm -it --cpus "0.5" ubuntu:22.04 bash
root@69aeb4ed7ca8:/# cat /sys/fs/cgroup/cpu.max
50000 100000
```

Reply:
I added two new test cases.

- `test_read_v2_sys_measurement_half_usage`
- `test_in_percentages_integration_v2_non_int_max`

The results show the behavior expected when there's a single CPU.